.github/autobuild/windows.ps1

```diff
@@ -0,0 +1,169 @@
+# Steps for generating Windows artifacts via Github Actions
+# See README.md in this folder for details.
+# See windows/deploy_windows.ps1 for standalone builds.
+
+param(
+    [Parameter(Mandatory=$true)]
+    [string] $Stage = "",
+    # Allow buildoption to be passed for jackonwindows build, leave empty for standard (ASIO) build:
+    [string] $BuildOption = "",
+    # unused, only required during refactoring as long as not all platforms have been updated:
+    [string] $GithubWorkspace =""
+)
+
+# Fail early on all errors
+$ErrorActionPreference = "Stop"
+
+$QtDir = 'C:\Qt'
+$ChocoCacheDir = 'C:\ChocoCache'
+$Qt32Version = "5.15.2"
+$Qt64Version = "5.15.2"
+$AqtinstallVersion = "2.0.6"
+$JackVersion = "1.9.17"
+$Msvc32Version = "win32_msvc2019"
+$Msvc64Version = "win64_msvc2019_64"
+$JomVersion = "1.1.2"
+
+$JamulusVersion = $Env:jamulus_buildversionstring
+if ($JamulusVersion -notmatch '^\d+\.\d+\.\d+.*')
+{
+    throw "Environment variable jamulus_buildversionstring has to be set to a valid version string"
+}
+
+Function Install-Qt
+{
+    param(
+        [string] $QtVersion,
+        [string] $QtArch
+    )
+    $Args = (
+        "--outputdir", "$QtDir",
+        "windows",
+        "desktop",
+        "$QtVersion",
+        "$QtArch",
+        "--archives", "qtbase", "qttools", "qttranslations", "qtwinextras"
+    )
+    aqt install-qt @Args
+    if (!$?)
+    {
+        echo "WARNING: Qt installation via first aqt run failed, re-starting with different base URL."
+        aqt install-qt -b https://mirrors.ocf.berkeley.edu/qt/ @Args
+        if (!$?)
+        {
+            throw "Qt installation with args @Args failed with exit code $LastExitCode"
+        }
+    }
+}
+
+Function Ensure-Qt
+{
+    if (Test-Path -Path $QtDir)
+    {
+        echo "Using Qt installation from previous run (actions/cache)"
+        return
+    }
+
+    echo "Install Qt..."
+    # Install Qt
+    pip install "aqtinstall==$AqtinstallVersion"
+    if (!$?)
+    {
+        throw "pip install aqtinstall failed with exit code $LastExitCode"
+    }
+
+    echo "Get Qt 64 bit..."
+    Install-Qt "${Qt64Version}" "${Msvc64Version}"
+
+    echo "Get Qt 32 bit..."
+    Install-Qt "${Qt32Version}" "${Msvc32Version}"
+}
+
+Function Ensure-jom
```

Comment:
I'm not sure if you used "Ensure-jom" because of [approved verbs](https://docs.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands?view=powershell-7.2) but I'd prefer something like Install-jom or similar. One could probably argue that it is named "Ensure" due to caching?

Reply:
Thanks for the pointer regarding approved verbs, I didn't know about it.

Yes, I chose that name due to caching. The function `Install-Qt` already existed and performs the actual installation. Therefore, I needed a name for the higher-level function which checks if Qt needs to be installed at all and only calls Install-Qt if necessary.

Ensure-jom and Ensure-JACK have been named to match that.

As it is, it is the most consistent across the stuff which we install.
I can change it to Install- for jom and JACK as actual installations do happen there (even on cached builds -- it's just that the installation source is from the cache). Do you think that would be better?

Ensure-Qt would still have to remain like that though (although I'm open to naming suggestions).

Reply:
I can’t come up with a better name at the moment. Probably it isn’t worth changing.